api/src/main/java/eu/coatrack/config/DeploymentMethodDependentConfig.java

```diff
@@ -0,0 +1,72 @@
+package eu.coatrack.config;
+
+/*-
+ * #%L
+ * coatrack-api
+ * %%
+ * Copyright (C) 2013 - 2022 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class DeploymentMethodDependentConfig {
+
+    /**
+     * URL of the "template jar" that CoatRack uses as basis to generate gateways.
+     * During development, this should usually point to your locally built coatrack gateway jar
+     * (but without the trailing version number and .jar filename extension, those will be added automatically).
+     */
+    public static final String GATEWAY_JAR_PATH_WITH_FILENAME_PREFIX;
+
+    /**
+     * The Web Application (admin) should store and manage the gateway config files in this directory while the
+     * config-server reads them from there.
+     */
+    public static final String GATEWAY_CONFIG_FILES_FOLDER_LOCATION;
+
+    static {
+        String isDeployedViaDockerCompose = System.getProperty("IS_DEPLOYED_VIA_DOCKER_COMPOSE");
+        if (isDeployedViaDockerCompose != null && isDeployedViaDockerCompose.equals("true")) {
+            GATEWAY_JAR_PATH_WITH_FILENAME_PREFIX = "file:///app/coatrack-proxy-";
+            GATEWAY_CONFIG_FILES_FOLDER_LOCATION = "/home/coatrack/gateway-config-files";
+        } else {
+            //Deployment is performed via CLI/IDE. The files are stored in/obtained from the coatrack project source folder.
+            GATEWAY_JAR_PATH_WITH_FILENAME_PREFIX = "file:///" + System.getProperty("user.dir") + "/spring-boot/proxy/target/coatrack-proxy-";
```

Comment:
This should not be hard-coded, this should stay a config parameter.

Reply:
Outdated.